modules/resourceid/fsw/src/cfe_resourceid_api.c

```diff
@@ -0,0 +1,133 @@
+/*
+**  GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**  Copyright (c) 2006-2019 United States Government as represented by
+**  the Administrator of the National Aeronautics and Space Administration.
+**  All Rights Reserved.
+**
+**  Licensed under the Apache License, Version 2.0 (the "License");
+**  you may not use this file except in compliance with the License.
+**  You may obtain a copy of the License at
+**
+**    http://www.apache.org/licenses/LICENSE-2.0
+**
+**  Unless required by applicable law or agreed to in writing, software
+**  distributed under the License is distributed on an "AS IS" BASIS,
+**  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**  See the License for the specific language governing permissions and
+**  limitations under the License.
+*/
+
+/*
+**  File:
+**  cfe_resource_api.c
+**
+**  Purpose:
+**      Function definitions related to CFE resource management
+**
+**  References:
+**     Flight Software Branch C Coding Standard Version 1.0a
+**     cFE Flight Software Application Developers Guide
+*/
+
+/*
+** Includes
+*/
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "cfe.h"
+#include "cfe_resourceid.h"
+#include "cfe_resourceid_basevalue.h"
+
+/*********************************************************************/
+/*
+ * CFE_ResourceId_GetBase
+ *
+ * For complete API information, see prototype in header
+ */
+uint32 CFE_ResourceId_GetBase(CFE_ResourceId_t ResourceId)
+{
+    return (CFE_ResourceId_ToInteger(ResourceId) & ~((uint32)CFE_RESOURCEID_MAX));
+}
+
+/*********************************************************************/
+/*
+ * CFE_ResourceId_GetSerial
+ *
+ * For complete API information, see prototype in header
+ */
+uint32 CFE_ResourceId_GetSerial(CFE_ResourceId_t ResourceId)
+{
+    return (CFE_ResourceId_ToInteger(ResourceId) & ((uint32)CFE_RESOURCEID_MAX));
+}
```

Comment:
The CFE_RESOURCEID_MAKE_BASE macro is defined in each option_inc resource_id.h file. These files are intended to be swappable pending user configuration, so the way bases are created might be different pending user definitions. Should base and serial conversions be handled there as well?

Reply:
Valid point, but the fundamental structure of the resource ID isn't really intended to be changeable by the user, it is a bitwise combination of a base and serial number, and other operations assume/require this general structure.  

What can change are the numeric values for the bases and the number of bits allocated to each.  (but currently the latter really should match OSAL at 16+16, for other reasons).

Reply:
Would be clarified if the configuration relative to OSAL is cleaned up

Reply:
Related to https://github.com/nasa/cFE/issues/1301